26 doc update documentation for UI setup and development (#29)
3PillarGlobal-Czechia/interview-app-ui: docs/API_CLIENT.md

# API Client

Communication with the backend API is implemented through the client in `ApiClient.ts`. It's an autogenerated file that contains all required methods and types for work with API. 

> Auto-generation is done through the package NSwag which uses open API documentation to generate typescript client code. NSwag is listed as a dev dependency in this project. 

Reviewer: `doc-link`

```suggestion
Auto-generation is done through the [NSwag package](https://openbase.com/js/nswag/documentation) which uses the open API documentation to generate typescript client code. NSwag is listed as a dev dependency in this project. 

```

`nswag.json` contains required configuration.

## Regeneration

> After any changes in the contract on API (new endpoints, URL changes, requests, or return types) ApiClinet needs to be regenerated to reflect all changes. 

Reviewer: `spelling/grammar`

```suggestion
After any changes to the API contract (new endpoints, URL changes, requests, or return types) ApiClient needs to be regenerated to reflect these changes. 

```
